File: backend/src/main/java/org/jugendhackt/online_klausuren/tasks/Task.java
package org.jugendhackt.online_klausuren.tasks;

import java.util.UUID;

public class Task {

    public final UUID id;
    public String type = "NONE";
    public final String title;
    public final String description;
    public final int time;

    public Task(String title, String description, int time, UUID id) {
        this.title = title;
        this.description = description;
        this.time = time;
        this.id = id;
    }
}

File: frontend/lib/model/task.dart
import 'package:online_klausuren_app/model/multiple_choice_task.dart';
import 'package:online_klausuren_app/model/text_task.dart';

class Task {
  String id;
  String type;
  String title;
  String description;
  int time;

  Task();

  toJson() {}

  factory Task.fromJson(Map<String, dynamic> json) {
    if (json['type'] == 'TEXT') {
      return TextTask.fromJson(json);
    } else if (json['type'] == 'CHOICES') {
      return MultipleChoiceTask.fromJson(json);
    } else {
      return Task.fromJson(json);
    }
  }
}
